frontend/src/pages/RecurringRunDetails.test.tsx

```diff
@@ -0,0 +1,400 @@
+    fullTestJob.enabled = false;
+    const tree = shallow(<RecurringRunDetails {...generateProps()} />);
+    TestUtils.makeErrorResponseOnce(enableJobSpy, 'could not enable');
+    await TestUtils.flushPromises();
+    const instance = tree.instance() as RecurringRunDetails;
+    const enableBtn = instance.getInitialToolbarState().actions.find(b => b.title === 'Enable');
+    await enableBtn!.action();
+    expect(updateDialogSpy).toHaveBeenCalledTimes(1);
+    expect(updateDialogSpy).toHaveBeenLastCalledWith(expect.objectContaining({
+      content: 'could not enable',
+      title: 'Failed to enable recurring run',
+    }));
+    tree.unmount();
+  });
+
+  it('calls enable API when enable button is clicked, refreshes the page', async () => {
+    fullTestJob.enabled = false;
+    const tree = shallow(<RecurringRunDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+    const instance = tree.instance() as RecurringRunDetails;
+    const enableBtn = instance.getInitialToolbarState().actions.find(b => b.title === 'Enable');
+    await enableBtn!.action();
+    expect(enableJobSpy).toHaveBeenCalledTimes(1);
+    expect(enableJobSpy).toHaveBeenLastCalledWith('test-job-id');
+    expect(getJobSpy).toHaveBeenCalledTimes(2);
+    expect(getJobSpy).toHaveBeenLastCalledWith('test-job-id');
+    tree.unmount();
+  });
+
+  it('shows a delete button', async () => {
+    const tree = shallow(<RecurringRunDetails {...generateProps()} />);
+    const instance = tree.instance() as RecurringRunDetails;
+    const deleteBtn = instance.getInitialToolbarState().actions.find(b => b.title === 'Refresh');
+    expect(deleteBtn).toBeDefined();
+    tree.unmount();
+  });
+
+  it('shows delete dialog when delete button is clicked', async () => {
+    const tree = shallow(<RecurringRunDetails {...generateProps()} />);
+    const instance = tree.instance() as RecurringRunDetails;
+    const deleteBtn = instance.getInitialToolbarState().actions.find(b => b.title === 'Delete');
+    await deleteBtn!.action();
+    expect(updateDialogSpy).toHaveBeenLastCalledWith(expect.objectContaining({
+      title: 'Delete this recurring run?',
+    }));
+    tree.unmount();
+  });
+
+  it('calls delete API when delete confirmation dialog button is clicked', async () => {
+    const tree = shallow(<RecurringRunDetails {...generateProps()} />);
+    await TestUtils.flushPromises();
+    const instance = tree.instance() as RecurringRunDetails;
+    const deleteBtn = instance.getInitialToolbarState().actions.find(b => b.title === 'Delete');
+    await deleteBtn!.action();
+    const call = updateDialogSpy.mock.calls[0][0];
+    const confirmBtn = call.buttons.find((b: any) => b.text === 'Delete');
+    await confirmBtn.onClick();
+    expect(deleteJobSpy).toHaveBeenCalledTimes(1);
+    expect(deleteJobSpy).toHaveBeenLastCalledWith('test-job-id');
+    tree.unmount();
+  });
+
+  it('does not call delete API when delete cancel dialog button is clicked', async () => {
```

Comment:
I remember from our discussion it wasn't easily doable, since it seems dismissing the dialog doesn't call its `onClose` callback. I'd rather put the effort to test this somewhere else. :)

Reply:
Ah right. Can you add a TODO here though? I've got one in the `NewRun` test so later we can find and clean these up

Reply:
Done.